sql/core/src/main/scala/org/apache/spark/sql/ArrowConverters.scala

```diff
@@ -18,27 +18,77 @@
 package org.apache.spark.sql
 
 import java.io.ByteArrayOutputStream
-import java.nio.channels.Channels
+import java.nio.ByteBuffer
+import java.nio.channels.{SeekableByteChannel, Channels}
 
 import scala.collection.JavaConverters._
 
 import io.netty.buffer.ArrowBuf
 import org.apache.arrow.memory.{BaseAllocator, RootAllocator}
 import org.apache.arrow.vector._
 import org.apache.arrow.vector.BaseValueVector.BaseMutator
-import org.apache.arrow.vector.file.ArrowWriter
+import org.apache.arrow.vector.file.{ArrowReader, ArrowWriter}
 import org.apache.arrow.vector.schema.{ArrowFieldNode, ArrowRecordBatch}
 import org.apache.arrow.vector.types.{FloatingPointPrecision, TimeUnit}
 import org.apache.arrow.vector.types.pojo.{ArrowType, Field, Schema}
 
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.types._
 
+
+//////////////////////
+private[sql] class ByteArrayReadableSeekableByteChannel(var byteArray: Array[Byte]) extends SeekableByteChannel {
```

Comment:
I copied this from somewhere in arrow test cases, is there an existing class that might work or maybe add to a public util package?